src/tests/ftest/container/SnapshotInfo.py

```diff
@@ -0,0 +1,550 @@
+            build_paths = json.load(finput)
+        basepath = os.path.normpath(build_paths['PREFIX']  + "/../")
+
+        server_group = self.params.get("server_group", '/server/',
+                                       'daos_server')
+
+        # setup the DAOS python API
+        self.context = DaosContext(build_paths['PREFIX'] + '/lib/')
+        self.d_log = DaosLog(self.context)
+
+        self.hostlist = self.params.get("test_machines", '/run/hosts/*')
+        hostfile = WriteHostFile.WriteHostFile(self.hostlist, self.workdir)
+
+        ServerUtils.runServer(hostfile, server_group, basepath)
+
+        # Set up the pool and container.
+        try:
+            # parameters used in pool create
+            createmode = self.params.get("mode", '/run/pool/createmode/')
+            createsetid = self.params.get("setname", '/run/pool/createset/')
+            createsize = self.params.get("size", '/run/pool/createsize/*')
+            createuid = os.geteuid()
+            creategid = os.getegid()
+
+            # initialize a pool object then create the underlying
+            # daos storage
+            self.log.info("==Creating a pool")
+            self.pool = DaosPool(self.context)
+            self.pool.create(createmode, createuid, creategid,
+                             createsize, createsetid, None)
+
+            # need a connection to create container
+            self.pool.connect(1 << 1)
+
+            # create a container
+            self.log.info("==Creating a container")
+            self.container = DaosContainer(self.context)
+            self.container.create(self.pool.handle)
+
+            # now open it
+            self.log.info("==Open the container")
+            self.container.open()
+
+        except DaosApiError as error:
+            self.log.info(error)
+            self.log.info(traceback.format_exc())
+            self.fail("##Test failed before snapshot taken")
+
+    def tearDown(self):
+        try:
+            if self.container:
+                self.log.info("==tearDown, close and destroy the container.")
+                self.container.close()
+                self.container.destroy()
+            if self.pool:
+                self.log.info("==tearDown, disconnect and destroy the pool.")
+                self.pool.disconnect()
+                self.pool.destroy(1)
+        finally:
+            self.log.info("==tearDown, stop the DAOS server.")
+            ServerUtils.stopServer()
+
+    def display_DaosContainer(self):
+        self.log.info("==display_DaosContainer========================")
+        self.log.info("self.container OBJ=                {0}"\
+            .format(self.container))
+        self.log.info("self.container.context obj=        {0}"\
+            .format(self.container.context))
+        self.log.info("self.container.context.libdaos=    {0}"\
+            .format(self.container.context.libdaos))
+        self.log.info("self.container.context.libtest=    {0}"\
```

Comment:
(pylint-logging-format-interpolation) Use % formatting in logging functions and pass the % parameters as arguments